lib/searchkick/query.rb

```diff
@@ -873,13 +873,11 @@ def custom_filter(field, value, factor)
 
     def boost_filters(boost_by, options = {})
       boost_by.map do |field, value|
-        log = value.key?(:log) ? value[:log] : options[:log]
-        value[:factor] ||= 1
         script_score = {
           field_value_factor: {
             field: field,
-            factor: value[:factor].to_f,
-            modifier: log ? "ln2p" : nil
+            modifier: (value[:modifier] or ( (value[:log] or options[:log]) ? 'ln2p' : nil ) or 'none'),
+            factor: (value[:factor] ? value[:factor].to_f : 1),
```

Comment:
The change for the factor field is purely aesthetic.

Reply:
Also since the `ln2p` was uses when either `value[:log]` or `options[:log]` are `true` a simple `or` statement was a better implementation. Additionally when there is no modifier set this will correctly default to `'none'` as per the ES documentation.

Reply:
And finally with this modification added I recommend that you deprecate the `log` option.

Reply:
The factor refactoring uses less memory this way and is a 1-liner. Can it stay?